operators/churner/tests/churner_test.go

```diff
@@ -96,30 +136,126 @@ func setupTest(t *testing.T) (*testbed.AnvilContainer, *testbed.LocalStackContai
 		_ = localstackContainer.Terminate(ctx)
 	})
 
-	return anvilContainer, localstackContainer, testConfig
+	return &TestSetup{
+		AnvilContainer:      anvilContainer,
+		LocalstackContainer: localstackContainer,
+		Contracts:           deploymentResult,
+		Operators:           operators,
+		PrivateKeys:         privateKeys,
+	}
 }
 
 func TestChurner(t *testing.T) {
 	ctx := t.Context()
-	_, _, testConfig := setupTest(t)
+	testSetup := setupTest(t)
+
+	// Create mock indexer
+	mockIndexer := &indexermock.MockIndexedChainState{}
+
+	// Start churner server directly using the churner package
+	grpcPort := "32002"
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%s", grpcPort))
+	require.NoError(t, err, "failed to listen on port")
+	defer func() {
+		if err := listener.Close(); err != nil {
+			t.Logf("failed to close listener: %v", err)
+		}
+	}()
```

Comment:
yet another place where we replicate all of this logic. This should all be abstracted in a churner struct with a Serve() method.

Reply:
Don't feel like doing it in this PR